src/embedded.test.js

```diff
@@ -0,0 +1,608 @@
+        });
+      });
+
+      test('appends "skip_domain_verification" to the iFrame URL if "skipDomainVerification" is valid', (done) => {
+        const client = new HelloSign();
+
+        client.on(HelloSign.events.OPEN, (data) => {
+          const url = new URL(data.iFrameUrl);
+
+          client.close();
+
+          expect(url.searchParams.has('skip_domain_verification')).toBe(true);
+          expect(url.searchParams.get('skip_domain_verification')).toBe('1');
+          done();
+        });
+
+        client.open(mockSignURL, {
+          clientId: mockClientId,
+          skipDomainVerification: true,
+        });
+      });
+
+      test('throws if "whiteLabeling" is not an object', () => {
+        const client = new HelloSign();
+
+        expect(() => {
+          client.open(mockSignURL, {
+            clientId: mockClientId,
+            whiteLabeling: 42,
+          });
+        }).toThrow(/"whiteLabeling" must be an object/);
+      });
+
+      test('appends "white_labeling_options" to the iFrame URL if "whiteLabeling" is valid', (done) => {
+        const client = new HelloSign();
+
+        client.on(HelloSign.events.OPEN, (data) => {
+          const url = new URL(data.iFrameUrl);
+
+          client.close();
+
+          expect(url.searchParams.has('white_labeling_options')).toBe(true);
+          expect(url.searchParams.get('white_labeling_options')).toBe(JSON.stringify({ foo: 'bar' }));
+          done();
+        });
+
+        client.open(mockSignURL, {
+          clientId: mockClientId,
+          whiteLabeling: {
+            foo: 'bar',
+          },
+        });
+      });
+    });
+
+    describe('close()', () => {
+
+      test('closes when the close button is clicked', (done) => {
+        const client = new HelloSign();
+
+        // Never visible when signing.
+        client.open(mockRequestURL, {
+          clientId: mockClientId,
+        });
+
+        const fn = jest.fn(() => {
+          expect(fn).toBeCalledTimes(1);
+          done();
+        });
+
+        client.once(HelloSign.events.CLOSE, fn);
+
+        const elem = client.element;
+        const closeBtn = elem.getElementsByClassName(settings.classNames.MODAL_CLOSE_BTN).item(0);
+
+        closeBtn.click();
+      });
+
+      test('emits the "close" event', (done) => {
+        const client = new HelloSign();
+
+        const fn = jest.fn(() => {
+          expect(fn).toBeCalledTimes(1);
+          done();
+        });
+
+        client.once(HelloSign.events.CLOSE, fn);
+
+        client.open(mockSignURL, {
+          clientId: mockClientId,
+        });
+
+        client.close();
+      });
+
+      test('does not attempt to close if not open', (done) => {
+        const client = new HelloSign();
+        const fn = jest.fn(() => {});
+
+        client.once(HelloSign.events.CLOSE, fn);
+
+        client.close();
+
+        setTimeout(() => {
+          expect(fn).toBeCalledTimes(0);
+          done();
+        }, 1000);
```

Comment:
I'm not a fan of this, but not sure how to do this better without using undocumented features like `done.fail()`. Jest needs better callback handling!

Reply:
Not really sure we even need this test /shrug

Reply:
Nixed